WordPress/Classes/Utility/CoreDataHelper.swift

```diff
@@ -360,3 +362,69 @@ extension CoreDataStack {
         try ContextManager.migrateDataModelsIfNecessary(storeURL: databaseLocation, objectModel: objectModel)
     }
 }
+
+/// This extension declares many `performQuery` usages that may introduce Core Data concurrency issues.
+///
+/// The context object used by the `performQuery` function is opaque to the caller. The caller should not assume what
+/// the context object is, nor the context queue type (the main queue or a background queue). That means the caller
+/// does not have enough information to guarantee safe access to the returned `NSManagedObject` instances.
+///
+/// The closure passed to the `performQuery` function should use the context to query objects and return none Core Data
```

Comment:
```suggestion
/// The closure passed to the `performQuery` function should use the context to query objects and return non- Core Data
```

nitpick: It sounds like the goal is to ensure the closure doesn't return Core Data types, so I think this wording could be tweaked, wdyt?

Reply:
Thanks for the suggestion! Taken and committed to https://github.com/wordpress-mobile/WordPress-iOS/pull/21008/commits/12554eaf066b819e0180c144fb314e6187d1eb1d